### DISCUSSION TITLE
虚拟机Ubuntu 20.04 搭建be调试环境，编译失败。

### DISCUSSION THREAD

**Discussion by stgztsw** [apache/doris#7393] [A - General / Q&A]
按照官方文档安装依赖包，在安装sudo apt install -y openssl-devel的时候失败，返回没有对应的依赖包。百度了一下说openssl-devel是centos的，Ubuntu可以openssl和libssl-dev，换成了openssl和libssl-dev后按照顺利。
之后再编译第三方依赖incubator-brpc-1.0.0-rc02的时候，出现了下面的报错，不知道如何解决

===== begin build incubator-brpc-1.0.0-rc02
CMake Deprecation Warning at CMakeLists.txt:18 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

-- The C compiler identification is GNU 10.3.0
-- The CXX compiler identification is GNU 10.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found gflags: /home/tangshiwei/branch/incubator-doris/thirdparty/installed/lib/libgflags.a
-- Found Protobuf: /home/tangshiwei/branch/incubator-doris/thirdparty/installed/lib64/libprotobuf.a;-lpthread (found version "3.14.0") 
-- Found Threads: TRUE  
-- Found OpenSSL: /home/tangshiwei/branch/incubator-doris/thirdparty/installed/lib64/libcrypto.a (found version "1.0.2k")  
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    GFLAGS_INCLUDE_DIR

-- Build files have been written to: /home/tangshiwei/branch/incubator-doris/thirdparty/src/incubator-brpc-1.0.0-rc02/doris_build
[1/335] Generating idl_options.pb.h, idl_options.pb.cc
FAILED: idl_options.pb.h idl_options.pb.cc 
cd /home/tangshiwei/branch/incubator-doris/thirdparty/src/incubator-brpc-1.0.0-rc02/doris_build && /home/tangshiwei/branch/incubator-doris/thirdparty/installed/bin/protoc -I/home/tangshiwei/branch/incubator-doris/thirdparty/installed/include -I/home/tangshiwei/branch/incubator-doris/thirdparty/src/incubator-brpc-1.0.0-rc02/src --cpp_out=/home/tangshiwei/branch/incubator-doris/thirdparty/src/incubator-brpc-1.0.0-rc02/doris_build /home/tangshiwei/branch/incubator-doris/thirdparty/src/incubator-brpc-1.0.0-rc02/src/idl_options.proto && /usr/local/bin/cmake -E copy /home/tangshiwei/branch/incubator-doris/thirdparty/src/incubator-brpc-1.0.0-rc02/doris_build/idl_options.pb.h /home/tangshiwei/branch/incubator-doris/thirdparty/src/incubator-brpc-1.0.0-rc02/doris_build/output/include/idl_options.pb.h
terminate called after throwing an instance of 'std::system_error'
  what():  Unknown error -1
Aborted (core dumped)
ninja: build stopped: subcommand failed.

**Comment by BiteTheDDDDt** [comment]
可以尝试在虚拟机里用docker编

**Reply by stgztsw** [reply]
谢谢回复啊，docker编译是可以的，这边试过了。但是我最终的目的是搭建be的调试环境来调试代码。请问在docker里面编译好的代码，可以在虚拟机里面用vscode调试吗？

**Reply by BiteTheDDDDt** [reply]
你在docker里编出来 是可以不依靠docker运行的
不过外面可能需要配一下jdk或者glibc什么的环境 可以尝试一下

**Reply by zyw8136** [reply]
这个是因为proto版本的问题，thirtyparty 文件夹里面的protoc是3.14.0,但是这个需要3.6.1的版本。

**Comment by dataalive** [comment]
you could use vscode to debug be code
